src/Administration/Resources/app/administration/src/app/filter.factory.js

```diff
@@ -0,0 +1,102 @@
+let filterFactory = {
+    /**
+     *
+     * @param {string} entityName
+     * @param {Object} filterSettings
+     * @returns {[]}
+     */
+    create(entityName, filterSettings) {
+        const entity = Shopware.EntityDefinition.get(entityName)
+
+        const filters = [];
+
+        for (const key in entity.properties) {
+            const property = entity.properties[key];
+            let settings = filterSettings[key];
+            if (settings === undefined) settings = {};
+            if (settings.hide) continue;
+
+            if (property.type === "uuid"
+                || property.type === "json_object"
+                || property.localField === "id"
+            ) {
+                continue;
+            }
+
+            const filter = {}
+
+            filter.name = key
+
+            if (settings.label) {
+                filter.label = settings.label;
+            } else {
+                filter.label = key; // this.$tc('sw-product.list.filter.manufacturer.label') // make translation
+            }
+
+            if (settings.placeholder) {
+                filter.placeholder = settings.placeholder;
+            } else {
+                filter.placeholder = ""; // this.$tc('sw-product.list.filter.manufacturer.placeholder') // make translation
+            }
+
+            if (property.localField) {
+                filter.field = entityName + '.' + property.localField
+            } else {
+                filter.field = entityName + '.' + key
+            }
+
+            let skip = false
+            switch (property.type) {
+                case "association":
```

Comment:
I would not do any type-mapping here. Simply check for the corresponding field-types while rendering the components. 

```
<sw-switch-field
                    v-if="filterOption.property.type === 'booeal'"
                    size="small"
                    :label="filterOption.label"
                    :placeholder="filterOption.placeholder"
                    v-model="filter[filterOption.property.name]">
            </sw-switch-field>
```
```